### PR TITLE
feat(metrics): Add failure_reason label to ingress_controller_configuration_push_count metric

### LINKED ISSUE #2484: Prometheus: `ingress_controller_configuration_push_count` should tell between intermittent errors and lock-ups
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem Statement

@seh reported that they'd like to use the `ingress_controller_configuration_push_count` Prometheus metric to alert in case a configuration lock-up (see #2195) happens, but that metric today does not tell between transient errors (e.g. network disconnect) and those that require fixing a config conflict (e.g. conflicting consumers #2324, #680)

### Proposed Solution

Add a metric label on the `ingress_controller_configuration_push_count` metric telling between failures that require fixing a conflict and those not requiring fixing a conflict.

### Additional information

Note that the existing metric `ingress_controller_translation_count[success=true|false]` may be answering this question already.

### Acceptance Criteria

- [x] Our prometheus integration includes a new label on the metric `ingress_controller_translation_count[success=true|false]` where true is successfully pushed configs while false is failed pushes
- [x] Our prometheus OOTB dashboard is updated with a new widget using the new label
- [x] Documentation is updated for the new metric/label 

_No response_

### PR DESCRIPTION
**Pull request by czeslavo**
**What this PR does / why we need it**:

* adds `failure_reason` label to the already existing `ingress_controller_configuration_push_count` metric which is reported every time configuration is pushed to the data plane 
* the label enables distinguishing config conflicts reasons: `conflict`, `network`, `other`. 

**Which issue this PR fixes**:

Part of #2484 

**Special notes for your reviewer**:

I was able to trigger configuration conflicts by adding consumers violating the unique username constraint. Admin API was responding with HTTP code `409`. I'm leveraging this, although it's not documented in Admin API's OpenAPI definitions. 

In general, it's hard to rely on deck to return any specific error therefore some error heuristics were needed.  

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

**Review by rainest — COMMENTED** [review]
How does this behave with these failures under database mode? Initial read suggests this will never mark a conflict with that, since it'll get back some sort of deck failure instead of any underlying go-kong failures.

We may not really be able to do anything yet given that the deck syncer doesn't always have the most useful errors, but a brief read indicates that we do get back an error array that we can walk to see if any are conflicts.

File: internal/metrics/prometheus.go [hunk 1]
```diff
@@ -52,9 +63,12 @@ func NewCtrlFuncMetrics() *CtrlFuncMetrics {
 				ProtocolKey + "` describes the configuration protocol (" + ProtocolDBLess + " or " +
 				ProtocolDeck + ") in use. `" +
 				SuccessKey + "` describes whether there were unrecoverable errors (`" +
-				SuccessFalse + "`) or not (`" + SuccessTrue + "`).",
+				SuccessFalse + "`) or not (`" + SuccessTrue + "`). `" +
+				FailureReasonKey + "` is populated in case of `" + SuccessKey +
+				"=\"" + SuccessFalse + "\"` " + "and describes the reason of failure (one of `" +
+				FailureReasonConflict + "`, `" + FailureReasonOther + "`).",
 		},
```
Comment (pmalek) [inline internal/metrics/prometheus.go:70]: Can we maybe rewrite this whole string formatting thingy into a nicely formatted `fmt.Sprintf()` invocation?

File: internal/dataplane/sendconfig/sendconfig_test.go [hunk 2]
```diff
@@ -128,3 +134,38 @@ func Test_updateReportingUtilities(t *testing.T) {
...
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+
+			reason := pushFailureReason(testCase.err)
+			require.Equal(t, testCase.expectedReason, reason)
+		})
+	}
```
Comment (pmalek) [inline internal/dataplane/sendconfig/sendconfig_test.go:170]:
```suggestion
	for _, tc := range testCases {
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
			t.Parallel()

			reason := pushFailureReason(tc.err)
			require.Equal(t, tc.expectedReason, reason)
		})
	}
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -51,6 +52,17 @@
  - [0.0.5](#005)
  - [0.0.4 and prior](#004-and-prior)
 
+## [2.7.0]
```
Comment (randmonkey) [inline CHANGELOG.md:55]: We are going to release 2.7.0 today. I think the changelog about what you have done should appear in 2.8.0.

**Comment by czeslavo** [comment]
> How does this behave with these failures under database mode? Initial read suggests this will never mark a conflict with that, since it'll get back some sort of deck failure instead of any underlying go-kong failures.
> 
> We may not really be able to do anything yet given that the deck syncer doesn't always have the most useful errors, but a brief read indicates that we do get back an error array that we can walk to see if any are conflicts.

I've tested that with Postgres mode and you were right, in case of `KongConsumer` username conflict, the failure comes from deck's `state.Get` or `dump.Get` which would have been missed without some additional heuristics: [2fbcb50](https://github.com/Kong/kubernetes-ingress-controller/pull/2965/commits/2fbcb50922200ec00dc30fb4a1f4bee100b49aca)

I extended that with detecting network errors specifically as well as it was cheap enough.

**Review by rainest — APPROVED** [review]
Moved changelog back to 2.7 with some minor edits. It's still stuck as I try and wait for E2E tests to slowly reveal if they're no longer dying due to environment failures. We already have a minor bump and this is pretty innocuous, so it may as well go out.

deck/format string changes look addressed, so :+1:

**Pull request merged**